### DISCUSSION TITLE
Apparent HttpClient timeout during transcription caused no error

### DISCUSSION THREAD

**Discussion by hiptopjones** [RageAgainstThePixel/OpenAI-DotNet#257] [Q&A]
# Bug Report

## Overview
I sent a roughly 90 minute audio file to the transcription endpoint, and it appeared to crash(?) the application silently.  The code is below.  The caller of this function writes the response text to a file, and that file was never written and no output was provided on the console.

If I manually throw an exception in here, I get loud and clear output on the console.

I had an inkling it was the default HttpClient timeout, so I injected my own HttpClient with a timeout of 600 seconds, and everything worked swimmingly.

I didn't see anything obvious in the EventViewer, which I would have expected if it was an actual crash.

The requests (especially the big ones) to OpenAI cost money, so I didn't try to reproduce it.  Filing the bug just in case you can see something from code inspection that would cause this.

Let me know if I can help further, or if you think I was doing something wrong.  I am using OpenAI-DotNet v7.7.4.

```csharp
        public async Task<string> Transcribe(string audioFilePath)
        {
            using var api = new OpenAIClient();

            var request = new AudioTranscriptionRequest(audioFilePath);
            var response = await api.AudioEndpoint.CreateTranscriptionAsync(request);
            return response;
        }
```

**Comment by StephenHodgson** [comment]
I think your file was too large. Try to find some silences in the clip and split it apart into easier to manage pieces.

It could have even been that the application ran out of memory in the upload buffer. But using large files like this probably is not the best strategy.

One other thing to try is to load the file into a memory stream, and pass that stream to the client. But I suspect it will have the same problem with memory constraint.

Unfortunately there is not a way to simultaneously stream, send and receive transcription data at the same time.

**Reply by hiptopjones** [reply]
As mentioned, I made it work by extending the HttpClient timeout.  The file was 7mb, so well below the 25mb limit of OpenAI’s API.

Since I got it working with a bigger timeout, I am good, but I filed the bug to register a possible problem for you in case silent exit on a reasonable size file with default HttpClient timeout and simple usage triggered an insight could save somebody else the trouble of debugging.

If I was less savvy, this might have been a blocker to using the library.

(BTW, thanks for the very useful library!)

**Reply by StephenHodgson** [reply]
Thanks. Typically, unless I can get a reproducible sample I don't usually consider it a bug until it can be verified.

There really shouldn't be any reason for it to silently exit, unless there was a misconfiguration of some kind.

You can also enable debugging on the client, to dump all responses to and from.

```csharp
using var client = new OpenAIClient { EnableDebug = true };
```
